cassandra-analytics-core/src/test/java/org/apache/cassandra/spark/bulkwriter/RecordWriterTest.java

```diff
@@ -74,8 +78,12 @@ class RecordWriterTest
 {
     private static final int REPLICA_COUNT = 3;
     private static final int FILES_PER_SSTABLE = 8;
-    // writing 270 rows with sstable size cap of 1 MB should produce 2 sstable
-    private static final int UPLOADED_SSTABLES = 2;
+    // writing 270 rows with sstable size cap of 1 MB should produce 2 sstables (Cassandra 4) or 3 sstables (Cassandra 5)
+    private static final Map<Integer, Integer> UPLOADED_SSTABLES_PER_CASSANDRA_VERSION = ImmutableMap.<Integer, Integer>builder()
+                                                                                                     .put(40, 2)
+                                                                                                     .put(41, 2)
+                                                                                                     .put(50, 3)
```

Comment:
I figure out why running with 50 outputs has 3 sstables. 
The key difference is at this method `org.apache.cassandra.spark.bulkwriter.TableSchemaTestCommon.MockTableInfoProvider#getCompression`. 

50 is using lz4. And 40 is using zstd. They make significant difference since the generated testing data is highly compressible.